packages/core/src/api.ts

```diff
@@ -1,14 +1,18 @@
-import { DsnLike } from '@sentry/types';
+import { DsnLike, SdkInfo } from '@sentry/types';
 import { Dsn, urlEncode } from '@sentry/utils';
 
 const SENTRY_API_VERSION = '7';
 
-/** Helper class to provide urls to different Sentry endpoints. */
+/**
+ * Helper class to provide urls, headers and metadata that can be used to form
+ * different types of requests to Sentry endpoints.
+ * Supports both envelopes and regular event requests.
+ **/
 export class API {
   /** The internally used Dsn object. */
   private readonly _dsnObject: Dsn;
   /** Create a new instance of API */
-  public constructor(public dsn: DsnLike) {
+  public constructor(public dsn: DsnLike, public metadata?: Partial<SdkInfo>) {
```

Comment:
Something like 

```suggestion
  public constructor(public dsn: DsnLike, public metadata?: {sdk: Partial<SdkInfo>}) {
```

if we leave room to have other metadata unrelated to sdkinfo?

Reply:
I'm +1 on this. We lose nothing, and gain future ability to add stuff as we need it.

Reply:
> if we leave room to have other metadata unrelated to sdkinfo?

That's why it's already a partial, to make it extensible in the future. If we change `SdkInfo` to `SdkMetadata` (just naming wise), we can treat it as a generic bag of data for the outgoing requests.

It also makes constructing whole pipeline more complex than it have to be right now, as the type for `Options` and `TransportOptions` would have to change from `metadata: Partial<SdkInfo>` to `metadata: { sdk: Partial<SdkInfo> }` as well. I'd much rather introduce `Metadata` type as `{ sdk: Partial<SdkInfo> }` and use it in every place.